packages/babel-preset/index.js

```diff
@@ -22,13 +22,21 @@ module.exports = api => ({
   plugins: [
     api.env('test')
       ? [
-          // This is needed to replace import.meta w/ process in Jest
+          // This is needed to replace import.meta.env w/ process in Jest
           // Jest does not play nicely w/ ESM and Vite uses import.meta
           // import.meta is only avaialable in ESM
           path.resolve(__dirname, 'importMetaEnvPlugin'),
         ]
-      : // The add-import-extension plugin causes Jest to error, but is needed for proper ESM builds
-        ['babel-plugin-add-import-extension'],
+      : false,
+    api.env('test')
+      ? [
+          // Also need to handle import.meta.url
+          'babel-plugin-transform-import-meta',
+        ]
+      : false,
+    api.env('test')
+      ? false // The add-import-extension plugin causes Jest to error, but is needed for proper ESM builds
+      : ['babel-plugin-add-import-extension'],
```

Comment:
Let's change this to something like the below so it's more readable/grouped by environment

```
api.env('test') ? [path.resolve(__dirname, 'importMetaEnvPlugin'), 'babel-plugin-transform-import-meta'] : false,
!api.env('test') ? ['babel-plugin-add-import-extension'] : false,
```

Reply:
I agree that is more readable, but it is _not_ the same thing. The [plugins config option](https://babeljs.io/docs/options#plugins) takes an array of type `Array<PluginEntry | Plugin>`. [PluginEntry is defined](https://babeljs.io/docs/options#pluginpreset-entries) as `EntryTarget | [EntryTarget, EntryOptions]`; doing as you suggest would pass in would result in:
```
plugins: [
  [ path.resolve(__dirname, 'importMetaEnvPlugin'), 'babel-plugin-transform-import-meta' ],
  ...
]
e.g. importing entry target `importMetaEnvPlugin` with the entry options `babel-plugin-transform-import-meta`. So then you get:
```
FAIL   @deephaven/code-studio  packages/code-studio/src/main/AppMainContainer.test.tsx
  ● Test suite failed to run

    [BABEL] /home/bender/dev/deephaven/iris-oss/web-client-ui/main/jest.setup.ts: .plugins[0][1] must be an object, false, or undefined
```

I tried using a spread operator, but using that in a ternary apparently screws things up as well:
![image](https://user-images.githubusercontent.com/4505624/231210261-295d2873-38f7-4348-b351-0d915ecb1c47.png)

I tried a few different ways, and this was the way that I hated the least. And I still hate it.

I guess I could put it into a function and that may be more readable, e.g.:
```
function getPlugins(isTestEnv) {
  if (isTestEnv) {
    return [
      // This is needed to replace import.meta.env w/ process in Jest
      // Jest does not play nicely w/ ESM and Vite uses import.meta
      // import.meta is only avaialable in ESM
      path.resolve(__dirname, 'importMetaEnvPlugin'),
      // Also need to handle import.meta.url
      'babel-plugin-transform-import-meta',
    ];
  }
  return [
    // The add-import-extension plugin causes Jest to error, but is needed for proper ESM builds
    'babel-plugin-add-import-extension',
  ];
}

...

  plugins: [
    ...getPlugins(api.env('test')),
    [
      'transform-rename-import',
      {
        // The babel-plugin-add-import-extension adds the .js to .scss imports, just convert them back to .css
        original: '^(.+?)\\.s?css.js$',
        replacement: '$1.css',
      },
    ],
  ].filter(Boolean),
```

... but I still hate it.

Reply:
I'll leave as is